enhancements/agent-installer/simplified-registry-less-cluster-operations.md

```diff
@@ -0,0 +1,426 @@
+release payload will be copied from the ISO to the node's disk (in the format
+of the registry's 'filesystem' backend) after the OS disk writing step - so
+that it remains available not only after the first reboot, but also once the
+installation is complete. Also the registry container image itself will be
+copied in the node additional image store for podman.
+[Assisted Service](https://github.com/openshift/assisted-service/)
+and [Assisted Installer](https://github.com/openshift/assisted-installer)
+will be enhanced accordingly to take care of this step, and to ensure it
+will be properly monitored during the installation process.
+
+A new registry service will be injected via MCO to support providing the
+release content stored on disk after the reboot (since the previous ephemeral
+environment will not be available anymore).
+
+To allow pulling images from the cluster, an ImageDigestMirrorSet
+(IDMS) policy will be added to the cluster manifests - using the `api-int` 
+DNS entry - so that the set of registries running within
+the control plane nodes will become the nominal pullspec of the current release
+payload itself.
+
+Finally, a new `InternalReleaseImage` custom resource, managed by MCO, will
+also be added as a cluster manifest to keep track and manage the currently
+stored internal release image.
+
+### Upgrade
+
+Upgrading an existing lower version cluster will be performed via the same
+extended ISO used for managing the installation. As soon as the user will
+attach the extended ISO to one of the control planes a temporary registry
+will be created to serve its content.
+At this point the user could edit the existing IRI resource to add the new
+release to trigger the copy process (from the temporary registry
+to each local control plane node registry), managed by the Machine Config
+Daemon (MCD). Once the copy has been completed on all the control plane nodes,
+the user could start the usual offline upgrade process via the `oc adm upgrade
+--allow-explicit-upgrade --to-image=<release-image>`.
+At the end of the upgrade the user may decide to delete the previous release
+entry from the IRI resource. In such case the MCD on each control plane node
+will take care of removing the older release payload from the local registry.
+
+### Adding a new node
+
+A node could be added to a target existing cluster using the `oc adm
+node-image` [command](https://github.com/openshift/enhancements/blob/master/enhancements/oc/day2-add-nodes.md)
+(currently, limited to workers). Even though the IDMS resource guarantees that
+the `oc adm node-image create` could successfully generate the node ISO, it
+won't be sufficient to support the joining process. For that, it will be
+necessary to run the registry on a host network port accessible from outside
+the cluster. In addition to that, the registry will be secured via a
+pull-secret. Users will be required to add this port to the api-int load
```

Comment:
The port should be configurable?
Or should we just guide users the open some constant port (e.g. 5000)?